cli/src/main/kotlin/com/westwinglabs/coinbase/cli/CoinbaseCli.kt

```diff
@@ -15,7 +15,7 @@ import org.apache.commons.cli.CommandLine
 import org.apache.logging.log4j.LogManager
 import java.io.File
 
-class CoinbaseCli : FeedListener() {
+open class CoinbaseCli : FeedListener() {
```

Comment:
Curious, why did you need to change the class to be `open`?

Reply:
1. I wanted to leverage all the methods that you created in `CoinbaseCli` without duplicating it
2. ...when all I really wanted to change was to overload the `onOpen` method in `CoinbaseCli` to take advantage of `SubscribeRequestAuthenticated`
3. ....but In Kotlin objects are closed by default. Essentially equivalent to objects created with `Final` in Java
4. So adding `open` to `CoinbaseCli` allowed me to create an anonymous object that inherited from it, which I could then overload its `onOpen` method [here](https://github.com/westwinglabs/coinbase-kotlin/pull/21/files#diff-f3b556d87f29468c42ead6aa82afa021ed65fed8e3faf85d1e0e3566ef8b069fR92).

Reply:
1. I wanted to leverage all the methods that you created in `CoinbaseCli` without duplicating it
2. ...when all I really wanted to change was to overload the `onOpen` method in `CoinbaseCli` to take advantage of `SubscribeRequestAuthenticated`
3. ....but In Kotlin objects are closed by default. Essentially equivalent to objects created with `Final` in Java
4. So adding `open` to `CoinbaseCli` allowed me to create an anonymous object that inherited from it, which I could then overload its `onOpen` method [here](https://github.com/westwinglabs/coinbase-kotlin/pull/21/files#diff-f3b556d87f29468c42ead6aa82afa021ed65fed8e3faf85d1e0e3566ef8b069fR92).